src/test/java/com/linkedin/kafka/clients/utils/tests/KafkaTestUtils.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.utils.tests;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.commons.io.FileUtils;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+
+
+public class KafkaTestUtils {
+  private final static AtomicBoolean SHUTDOWN_HOOK_INSTALLED = new AtomicBoolean(false);
+  private final static Thread SHUTDOWN_HOOK;
+  private final static List<File> FILES_TO_CLEAN_UP = Collections.synchronizedList(new ArrayList<>());
+
+  static {
+    SHUTDOWN_HOOK = new Thread(() -> {
```

Comment:
Is this just here just in case tear down does not work?

Reply:
no. closable test classes (like embeddable brokers and zk) should delete their files on close, but other tests may need files/folders